File: Data Structures & Algorithms/house-robber/submission-0.java
class Solution {
    Integer dp[];
    public int find(int i,int a[]){
        if(i>=a.length) return 0;
        if(dp[i]!=null) return dp[i];
        int take=a[i]+find(i+2,a);
        int skip=find(i+1,a);
        return dp[i]=Math.max(take,skip);
    }
    public int rob(int[] nums) {
        dp=new Integer[nums.length];
        return find(0,nums);
        
    }
}

File: Data Structures & Algorithms/max-consecutive-ones/submission-0.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        ma=0
        r=0
        for i in nums:
            if i==1:
                r+=1
            else:
                ma=max(r,ma)
                r=0
        return max(r,ma)

        